Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -477,25 +510,36 @@ bool CSCGeometryParsFromDD::build(const cms::DDCompactView* cview,
       std::transform(
           dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
 
-      fpar.emplace_back((dpar[1]));
-      fpar.emplace_back((dpar[2]));
-      fpar.emplace_back((dpar[3]));
-      fpar.emplace_back((dpar[4]));
+      fpar.emplace_back((dpar[1] / dd4hep::cm));
+      fpar.emplace_back((dpar[2] / dd4hep::cm));
+      fpar.emplace_back((dpar[3] / dd4hep::cm));
+      fpar.emplace_back((dpar[4] / dd4hep::cm));
+      edm::LogVerbatim("CSCGeometryParsFromDD")
+          << "(7) - Subtraction - dpar[1] (ddd dpar[4]): " << dpar[1] / dd4hep::cm
+          << " (ddd dpar[8]):  " << dpar[2] / dd4hep::cm << " dpar[3] (as ddd): " << dpar[3] / dd4hep::cm
+          << " dpar[4] (ddd dpar[0]): " << dpar[4] / dd4hep::cm;
     } else {
       dpar = fv.parameters();
 
       std::transform(
           dpar.begin(), dpar.end(), dpar.begin(), [](double i) -> double { return cms_rounding::roundIfNear0(i); });
 
-      fpar.emplace_back((dpar[0]));
-      fpar.emplace_back((dpar[1]));
-      fpar.emplace_back((dpar[2]));
-      fpar.emplace_back((dpar[3]));
+      fpar.emplace_back((dpar[0] / dd4hep::cm));
+      fpar.emplace_back((dpar[1] / dd4hep::cm));
+      fpar.emplace_back((dpar[2] / dd4hep::cm));
+      fpar.emplace_back((dpar[3] / dd4hep::cm));
+      edm::LogVerbatim("CSCGeometryParsFromDD")
+          << "(7)Bis - Else - dpar[0]: " << dpar[0] / dd4hep::cm << " dpar[1]: " << dpar[1] / dd4hep::cm
+          << " dpar[2]:  " << dpar[2] / dd4hep::cm << " dpar[3]: " << dpar[3] / dd4hep::cm;
     }
 
-    gtran[0] = static_cast<float>(fv.translation().X());
-    gtran[1] = static_cast<float>(fv.translation().Y());
-    gtran[2] = static_cast<float>(fv.translation().Z());
+    gtran[0] = static_cast<float>(fv.translation().X() / dd4hep::cm);
+    gtran[1] = static_cast<float>(fv.translation().Y() / dd4hep::cm);
+    gtran[2] = static_cast<float>(fv.translation().Z() / dd4hep::cm);
+
+    edm::LogVerbatim("CSCGeometryParsFromDD")
+        << "(8) gtran[0]: " << gtran[0] / dd4hep::cm << " gtran[1]: " << gtran[1] / dd4hep::cm
+        << " gtran[2]: " << gtran[2] / dd4hep::cm;
```

Comment:
I noticed lines 557-562 contain a useless loop. Those lines can be changed to:
```
 if (wg.numberOfGroups == 0) {
       LogTrace(myName) << myName << " wg.numberOfGroups == 0 ";
    }
```

Reply:
@cvuosalo : done